05.wc/wc.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, file_paths = parse_options(ARGV)
+
+  # パイプで接続した場合 $stdin.tty? は false になる
+  if file_paths.empty? && !$stdin.tty?
+    file_stats = count_content($stdin.read)
+    file_stats[:file_path] = nil
+    file_stats_list = [file_stats]
+    total_stats = nil
+  else
+    file_stats_list, total_stats = collect_file_stats(file_paths)
+  end
+
+  max_width = calculate_max_width(file_stats_list, total_stats)
+
+  file_stats_list.each { |stats| output_format(stats, options, max_width) }
+
+  output_format(total_stats, options, max_width) if total_stats
```

Comment:
```suggestion
  output_format(total_stats, options, max_width) if file_stats_list.size > 1
```

「total_statsがnilであること」に特別な意味を持たせるよりも、「データが2件以上あれば」という条件にした方が仕様を明確に表現できそうな気がします。

Reply:
確かにその表現の方が意味が伝わりやすいですね。修正しました。
https://github.com/cellotak/ruby-practices/pull/12/commits/f5784daf39f84312a4f790c72dbebb600de56ce2